web/ui/react-app/src/pages/Status.tsx

```diff
@@ -73,36 +84,40 @@ const Status = () => {
       />
     );
   }
-  return data
-    ? data.map((statuses, i) => {
-        return (
-          <Fragment key={i}>
-            <h2>{sectionTitles[i]}</h2>
-            <Table className="h-auto" size="sm" bordered striped>
-              <tbody>
-                {Object.entries(statuses).map(([k, v]) => {
-                  const { title = k, customizeValue = (val: any) => val, customRow, skip } = statusConfig[k] || {};
-                  if (skip) {
-                    return null;
-                  }
-                  if (customRow) {
-                    return customizeValue(v);
-                  }
-                  return (
-                    <tr key={k}>
-                      <th className="capitalize-title" style={{ width: '35%' }}>
-                        {title}
-                      </th>
-                      <td className="text-break">{customizeValue(v)}</td>
-                    </tr>
-                  );
-                })}
-              </tbody>
-            </Table>
-          </Fragment>
-        );
-      })
-    : null;
+  return (
+    <>
+      {data
+        ? data.map((statuses, i) => {
+            return (
+              <Fragment key={i}>
+                <h2>{sectionTitles[i]}</h2>
+                <Table className="h-auto" size="sm" bordered striped>
+                  <tbody>
+                    {Object.entries(statuses).map(([k, v]) => {
+                      const { title = k, customizeValue = (val: any) => val, customRow, skip } = statusConfig[k] || {};
+                      if (skip) {
+                        return null;
+                      }
+                      if (customRow) {
+                        return customizeValue(v);
+                      }
+                      return (
+                        <tr key={k}>
+                          <th className="capitalize-title" style={{ width: '35%' }}>
+                            {title}
+                          </th>
+                          <td className="text-break">{customizeValue(v)}</td>
+                        </tr>
+                      );
+                    })}
+                  </tbody>
+                </Table>
+              </Fragment>
+            );
+          })
+        : null}
+    </>
+  );
```

Comment:
`data.map()` returns an array of React elements, which TypeScript doesn't allow for function components (the function wasn't properly typed before). I thought TypeScript also didn't allow returning nulls from function components (the TS error message was... let's say hard to read), but this is not actually the case, only the `data.map()` portion was the problem.

I changed it now so that the fragment only encloses the array, not the null.

Reply:
```const Status = ({ pathPrefix = '' }: RouteComponentProps & PathPrefixProps) => {```
I think this will eliminate fragments and the component will remain properly typed

Reply:
Then I get the following error in `App.tsx` because it's not fully typed as a function component anymore:

`JSX element type 'Element | Element[] | null' is not a constructor function for JSX elements.
  Type 'Element[]' is missing the following properties from type 'Element': type, props, key`

I think adding the one extra `<></>` is fine though, and with `FC<...>` it's clearer what it is.

Removing `data ?` I get `Object is possibly 'undefined'.`. I could instead do `data!`, but not sure if that's nicer.

Reply:
I think the extra `<></>` is fine, I just wanted to understand the need for it and now I do, thanks!